src/main/java/io/orangebeard/listener/OrangebeardTestSystemListener.java

```diff
@@ -185,25 +187,28 @@ public void testComplete(TestPage testPage, TestSummary testSummary) {
     }
 
     private void logScenarioLibraries(UUID testUUID, List<WikiPage> scenarioLibraries) {
-        for (WikiPage scenarioLibrary : scenarioLibraries) {
-            String log = scenarioLibrary.getHtml();
+        if (orangebeardProperties.logShouldBeDispatchedToOrangebeard(LogLevel.debug)) {
 
-            log = OrangebeardTableLogParser.applyOrangebeardTableStyling(log);
-            String enrichedLog = OrangebeardTableLogParser.embedImagesAndStripHyperlinks(log, rootPath);
+            for (WikiPage scenarioLibrary : scenarioLibraries) {
+                String log = scenarioLibrary.getHtml();
 
-            //Workaround for corner case where table contains binary representation with 0x00 unicode chars
-            enrichedLog = enrichedLog.replace("\u0000", "");
+                log = OrangebeardTableLogParser.applyOrangebeardTableStyling(log);
+                String enrichedLog = OrangebeardTableLogParser.embedImagesAndStripHyperlinks(log, rootPath);
 
-            Log logItem = Log.builder()
-                    .message(enrichedLog)
-                    .itemUuid(testUUID)
-                    .testRunUUID(runContext.getTestRunUUID())
-                    .logLevel(LogLevel.debug)
-                    .time(LocalDateTime.now())
-                    .build();
+                //Workaround for corner case where table contains binary representation with 0x00 unicode chars
+                enrichedLog = enrichedLog.replace("\u0000", "");
 
-            orangebeardClient.log(logItem);
-            numberOfLogs++;
+                Log logItem = Log.builder()
```

Comment:
We can theoretically win some more perf here by queuing multiple scenariolibraries into one call?

Reply:
We could, but that's not in scope of this pull request.